test/AutoRest.TestServer.Tests/NumberTest.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using System.Threading.Tasks;
+using AutoRest.TestServer.Tests.Infrastructure;
+using body_number;
+using NUnit.Framework;
+
+namespace AutoRest.TestServer.Tests
+{
+    public class NumberTest : TestServerTestBase
+    {
+        public NumberTest(TestServerVersion version) : base(version, "number") { }
+
+        [Test]
+        public Task putFloatBigScientificNotation() => TestStatus(async (host, pipeline) => await NumberOperations.PutBigFloatAsync(ClientDiagnostics, pipeline, 3.402823e+20f, host));
+
+        [Test]
+        public Task putFloatSmallScientificNotation() => TestStatus(async (host, pipeline) => await NumberOperations.PutSmallFloatAsync(ClientDiagnostics, pipeline, 3.402823e-20f, host));
+
+        [Test]
+        public Task putDoubleSmallScientificNotation() => TestStatus(async (host, pipeline) => await NumberOperations.PutSmallDoubleAsync(ClientDiagnostics, pipeline, 2.5976931e-101d, host));
+
+        [Test]
+        public Task putDoubleBigScientificNotation() => TestStatus(async (host, pipeline) => await NumberOperations.PutBigDoubleAsync(ClientDiagnostics, pipeline, 2.5976931e+101d, host));
+
+        [Test]
+        [IgnoreOnTestServer(TestServerVersion.V2, "Match too strict")]
+        public Task putDecimalBigPositiveDecimal() => TestStatus(async (host, pipeline) => await NumberOperations.PutBigDecimalPositiveDecimalAsync(ClientDiagnostics, pipeline, host));
+
+        [Test]
+        [Ignore("Value 2.5976931e+101 is out of range of C# decimal")]
```

Comment:
Hmm. Do you think we need to use a different type than `decimal`? Or, would we just change the tests on the test server?

Reply:
This is an optional test, I'll file an issue to follow up

Reply:
https://github.com/Azure/autorest.csharp/issues/335